packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +476,45 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("should not call entry point before data store is requested", async () => {
+		const dataStoreFactory = new TestFluidObjectFactory();
+		const dataStoreContext = new TestDataObjectContext();
+
+		assert.doesNotThrow(
+			async () => dataStoreFactory.instantiateDataStore(dataStoreContext, false),
+			"entry point should not be called during initizalization.",
+		);
+
+		// Only call entry point when container is requested, verify error thrown if not.
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory.type, Promise.resolve(dataStoreFactory)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+			{},
+		);
+		const _createContainer = async (): Promise<IContainer> => {
+			return provider.createContainer(runtimeFactory);
+		};
+		const container = await _createContainer();
+		await assert.rejects(
```

Comment:
I am not sure I understand what this is validating either. It seems its validating that the test factory's initialization should not be called?

Reply:
And there is verifying the enttryPoint function is called when container make a request to data store. Entrypoint function is throwing error so the promise should be rejected and error message is the one we defined in entrypoint function.

Reply:
But this is an expected behavior that if request is explicitly called, the entry point will be loaded. Again, you should not be using mocks to test this behavior.